src/reconstruction/StandardModel.jl

```diff
@@ -204,19 +214,24 @@ end
 end
 
 @_remove_fn metabolite StandardModel String inplace begin
-    remove_metabolites!(model, [metabolite_id])
+    if !(metabolite_id in metabolites(model))
+        @_models_log @info "Metabolite $metabolite_id not found in model."
+    else
+        remove_reactions!(
+            model,
+            [
+                rid for
+                (rid, rn) in model.reactions if any(haskey(rn.metabolites, metabolite_id))
+            ],
+        )
+        delete!(model.metabolites, metabolite_id)
+    end
+    nothing
 end
 
 @_remove_fn metabolite StandardModel String inplace plural begin
-    remove_reactions!(
-        model,
-        [
-            rid for (rid, rn) in model.reactions if
-            any(haskey.(Ref(rn.metabolites), metabolite_ids))
-        ],
-    )
-    delete!.(Ref(model.metabolites), metabolite_ids)
-    return nothing
+    remove_metabolite!.(Ref(model), metabolite_ids)
+    nothing
```

Comment:
why do this in small pieces? The original impl is roughly O(nnz(S)), the new one is O(nnz(S)*length(metabolite_ids))